qa_python_2 with 8 tests for BooksCollector class

FruitHuntress opened the pull request.

FruitHuntress committed:
qa_python_2 with 8 tests for BooksCollector class

AlexKitIt commented on tests.py:
```diff
@@ -21,4 +21,74 @@ def test_add_new_book_add_two_books(self):
+
+
+
+    #1 Проверка, что нельзя добавить уже имеющуюся книгу в списке
+    def test_add_new_book_add_existing_book(self):
```
Можно лучше: пометить каждый тест комментом с описанием, что именно он делает - хорошее решение, но это понятно из нейминга метода (тут ты молодец), и это несколько засоряет код. Гораздо полезнее было бы написать к нему докстринг (если уж действительно нужно пояснить как он работает), чтобы получать этот коммент в любом месте кода, где ты будешь впоследствии вызывать этот метод. Коммент рекомендательный

AlexKitIt commented on tests.py:
```diff
@@ -21,4 +21,74 @@ def test_add_new_book_add_two_books(self):
+
+        collector.add_new_book('Гордость и предубеждение и зомби')
+        collector.add_new_book('Гордость и предубеждение и зомби')
+        assert len(collector.get_books_rating()) == 1
```
Можно лучше: написание assert message после вызова самого assert значительно упросит тебе потом процесс отладки. Коммент рекомендательный

AlexKitIt commented on tests.py:
```diff
@@ -21,4 +21,74 @@ def test_add_new_book_add_two_books(self):
+        collector = BooksCollector()
+
+        collector.add_new_book('Гордость и предубеждение и зомби')
+        assert collector.get_books_rating()
```
Можно лучше: тут мы не узнаем, что вернулся именно словарь, тут мы узнаем, что вернется не None,не False, и не какой-то пустой объект (по типу пустого словаря или строки) В этом кейсе лучше бы проверить вхождение заданной книги в состав ключей словаря

AlexKitIt commented on tests.py:
```diff
@@ -21,4 +21,74 @@ def test_add_new_book_add_two_books(self):
+    def test_get_list_of_favorites_books_with_zero_books_added(self):
+        collector = BooksCollector()
+
+        assert collector.get_list_of_favorites_books() == []
```
Нужно исправить: не покрыты:
- значения рейтингов, выходящие за заданный диапазон (с 1 до 10)
- добавление в избранное книги, не содержащейся в books_rating 
- нельзя выставить рейтинг книге, которой нет в списке.
К заданию шла подсказка с кейсами,которые стоит реализовать, посмотри ее еще раз. Негативные кейсы тоже нужно проверять)

FruitHuntress committed:
added tests:
test_set_book_rating_set_rating_more_than10
test_set_book_rating_set_rating_zero
test_set_book_rating_book_off_the_list_rating_is_not_set
test_add_book_in_favorites_book_off_the_dict_is_not_added
